docs/src/operate/security/cumulocity_token.md

```diff
@@ -0,0 +1,175 @@
+
+1. Subscribe to the token topic
+
+    ```sh te2mqtt formats=v1
+    tedge mqtt sub c8y/s/dat --no-topic
+    ```
+
+2. Publish an empty message on the `c8y/s/uat` topic
+
+    ```sh te2mqtt formats=v1
+    tedge mqtt pub c8y/s/uat ''
+    ```
+
+3. After a while the token will be published on the subscribed topic `c8y/s/dat` in the below format
+
+    ```sh
+    71,${Base64 encoded JWT}
+    ```
+
+    Store the token as required (e.g. assign to a variable), and use in REST API request to the Cumulocity IoT tenant.
+
+    :::note
+    Typically tokens are only valid for 1 hour (depending on your Cumulocity IoT settings), so you will need to request a new token before it expires otherwise your API Requests will return an Unauthorized (401) error.
+
+    The expiration timestamp, issuer (Cumulocity IoT URL), and tenant are encoded in the token and can be decoded using the standard [JSON Web Token Standard](https://datatracker.ietf.org/doc/html/rfc7519) which there should be a library in most popular programming languages.
+    :::
+
+### Alternative: Retrieving the token using mosquitto_rr
+
+`mosquitto_rr` provides a simple cli interface to handle the request/response pattern, and can be used to create a simple one-liner to retrieve a new token.
+
+The `mosquitto_rr` cli command can be installed on your operating system via one of the following packages:
+
+```sh tab={"label":"Debian/Ubuntu"}
+mosquitto-clients
+```
+
+```sh tab={"label":"RHEL/Fedora/RockyLinux"}
+mosquitto
+```
+
+```sh tab={"label":"openSUSE"}
+mosquitto-clients
+```
+
+```sh tab={"label":"Alpine"}
+mosquitto
+```
+
+The token can be retrieved using the following one-liner:
+
+```sh
+export C8Y_TOKEN=$(mosquitto_rr -t c8y/s/uat -e c8y/s/dat -m '' | cut -d, -f2-)
+```
+
+Where:
+* `-t` represents the request topic, where the `-m ''` message is sent to it
+* `-e` represents the response topic which will print the message on the console
+
+
+## Using token in REST API calls to Cumulocity IoT
+
+The retrieved token can be used to make HTTP calls to the [Cumulocity IoT REST API](https://cumulocity.com/api/core/).
+
+For simplicity, this example will retrieve the Cumulocity IoT URL via the tedge cli command. If you are using a higher level programming language like python3, then you get the Cumulocity IoT URL by decoding the token, e.g. using [PyJWT](https://pyjwt.readthedocs.io/en/latest/).
+
+The following code snippet does the following steps:
+
+1. Get the Cumulocity IoT URL (using `tedge`)
+2. Get the token (using `mosquitto_rr`)
+3. Send a request (using `curl`)
+
+```sh
+# Get the Cumulocity IoT URL
+export C8Y_URL="https://$(tedge config get c8y.url)"
+
+# Get the token (using mosquitto_rr cli command)
+export C8Y_TOKEN=$(mosquitto_rr -t c8y/s/uat -e c8y/s/dat -m '' | cut -d, -f2-)
+
+# Send a REST API call to Cumulocity IoT
+curl -s -H "Authorization: Bearer $C8Y_TOKEN" \
+    -H "Accept: application/json" \
+    "$C8Y_URL/user/currentUser"
```

Comment:
As a conclusion of this example, I would add the equivalent call using the Cumulocity Proxy. just to make the reader want.

```
curl http://localhost:8001/c8y/user/currentUser 
```

Reply:
Resolved by 93b12f47b181b42cd9709e8639f349722ec8d152

Reply:
Resolved by 93b12f47b181b42cd9709e8639f349722ec8d152